evaluation/e2e_samples/chunking/notebook.ipynb

```diff
@@ -0,0 +1,584 @@
+    "            algorithms=[\n",
+    "                HnswVectorSearchAlgorithmConfiguration(\n",
+    "                    name=_HNSW_ALGORITHM_CONFIG_NAME,\n",
+    "                    kind=VectorSearchAlgorithmKind.HNSW,\n",
+    "                    parameters=HnswParameters(\n",
+    "                        m=_M,\n",
+    "                        ef_construction=_EF_CONSTRUCTION,\n",
+    "                        ef_search=_EF_SEARCH,\n",
+    "                        metric=_METRIC\n",
+    "                    )\n",
+    "                )\n",
+    "            ],\n",
+    "            profiles=[\n",
+    "                VectorSearchProfile(\n",
+    "                    name=_VECTOR_SEARCH_PROFILE_NAME,\n",
+    "                    algorithm=_HNSW_ALGORITHM_CONFIG_NAME\n",
+    "                )\n",
+    "            ]  \n",
+    "        )\n",
+    "\n",
+    "        semantic_settings: SemanticSettings | None = None\n",
+    "        if with_semantic_search:\n",
+    "            semantic_settings = SemanticSettings(\n",
+    "                configuration=[\n",
+    "                    SemanticConfiguration(\n",
+    "                        name=_DEFAULT_SEMANTIC_CONFIG_NAME,\n",
+    "                        prioritized_fields=PrioritizedFields(\n",
+    "                            prioritized_content_field=[\n",
+    "                                SemanticField(field_name=_CONTENT_FIELD),\n",
+    "                                SemanticField(field_name=_METADATA_FIELD)\n",
+    "                            ]\n",
+    "                        )\n",
+    "                    )\n",
+    "                ]\n",
+    "            )\n",
+    "\n",
+    "        index = SearchIndex(\n",
+    "            name=index_name,\n",
+    "            fields=fields,\n",
+    "            vector_search=vector_search,\n",
+    "            semantic_settings=semantic_settings)\n",
+    "        return self._search_index_client.create_index(index)\n",
+    "    \n",
+    "    def get_or_create_index_client(\n",
+    "        self,\n",
+    "        index_name: str,\n",
+    "        with_semantic_search: bool = False\n",
+    "    ):\n",
+    "        try:\n",
+    "            return self._get_index_client(index_name)\n",
+    "        except ResourceNotFoundError:\n",
+    "            print(f\"Index {index_name} does not exist... Creating a new index.\")\n",
+    "            return self._create_index_client(index_name, with_semantic_search)\n",
+    "\n",
+    "    def delete_index(self, index_name: str):\n",
+    "        self._search_index_client.delete_index(index_name)\n",
+    "\n",
+    "    @classmethod\n",
+    "    def from_config(cls, config: AzureAiSearchConfig):\n",
+    "        search_index_client = SearchIndexClient(config.api_endpoint, AzureKeyCredential(config.api_key))\n",
+    "        return cls(search_index_client)\n",
+    "\n",
+    "\n",
+    "class GroundTruthDatasetLoader(object):\n",
```

Comment:
not sure I understand the need for this class abstraction around these pandas DF methods, i think it takes away from the clarity of the run method in the ExperimentOrchestrator class.

don't think we need to address that in this PR, but could we get a discussion in this thread explaining it so that we can consider any future changes to this section?

Reply:
We never pass is in to the ExperimentOrchestrator's from_config method, but the idea is to allow people to substitute the data loader if they wanted to. I think the organization of the code would have been better if it wasn't in a notebook, so we could just do dataframe operations directly in the orchetrator if we wanted to.